### DISCUSSION TITLE
How to compile in the yocto project?

### DISCUSSION THREAD

**Discussion by zuowanbushiwo** [roc-streaming/roc-toolkit#286] [Q&A] [answered]
The Yocto project supports the compilation of the scons project, how to write a bitbake file, so that the entire project is compiled in Yocto like mongodb (https://github.com/Anand1985-gl/my-meta-oe/blob/master/meta-oe /recipes-dbs/mongodb/mongodb_git.bb) Same?

thanks!

**Comment by gavv** [comment, marked as answer]
Hi,

I didn't use Yocto before and didn't try to build Roc for it.

Generally speaking, I guess you'll need:

* to specify dependencies; you can find them here: https://roc-project.github.io/roc/docs/building/dependencies.html

* to specify a download url; you can find it here: https://github.com/roc-project/roc/releases

* to specify build command and options; you can find them here: https://roc-project.github.io/roc/docs/building/user_cookbook.html and here: https://roc-project.github.io/roc/docs/building/scons_options.html

If you need to cross-compile Roc, you can also read this article: https://roc-project.github.io/roc/docs/portability/cross_compiling.html

**Comment by zuowanbushiwo** [comment]
thanks!
